src/components/PopupWithForm.js

```diff
@@ -7,17 +7,25 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputSelector = config.inputSelector;
-    this._submitButtonSelector = config.submitButtonSelector;
-    //this._submitButton = this._element.querySelector(".modal__button");
+
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+
+    this._defaultButtonText = this._submitButton.textContent = "Save"
+    
+    //this._submitButtonSelector = config.submitButtonSelector;
+    //this._submitButtonSelector.textContent = savingProcess;
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`